osquery/core/init.cpp

```diff
@@ -486,8 +491,6 @@ void Initializer::initShell() const {
 void Initializer::initWatcher() const {
   // The watcher should not log into or use a persistent database.
   if (isWatcher()) {
-    FLAGS_disable_database = true;
-    FLAGS_disable_logging = true;
```

Comment:
flag overwriting is moved to the top so that initStatusLogger will know for sure if FLAGS_disable_logging is set to true or not.

Because of that knowledge, initStatusLogger can now setup  BufferedLogSink::get().setUp() only when logging is really allowed.